app/src/ui/discard-changes/discard-changes-dialog.tsx

```diff
@@ -57,11 +58,19 @@ export class DiscardChanges extends React.Component<
   }
 
   public render() {
+    const discardingAllChanges = this.props.discardingAllChanges
+
     return (
       <Dialog
         id="discard-changes"
         title={
-          __DARWIN__ ? 'Confirm Discard Changes' : 'Confirm discard changes'
```

Comment:
i looked and the components in this codebase are littered with this logic. would you be up for creating a helper function that takes a string and returns the platform appropriate casing of it? and using it here? (no need to replace all instances of that pattern in the codebase—just having the helper function for future use would be amazing!)

Reply:
@Daniel-McCarthy exactly! so something like function that does: 
`string` -> `string in the current platform's casing`

Reply:
Okay, I got a chance to look into this. I have a solution that can emulate what I believe we're going for.
```typescript
  /** Convert input string to current platform's text case preference. */
  private toPlatformCase(inputText: string) {
    inputText = inputText.toLowerCase()

    if (__DARWIN__) {
      // Capitalize the first letter of every word.
      inputText = inputText.replace(/\b[a-z]/gi, $1 => $1.toUpperCase())
    } else {
      // Capitalize the first letter of the first word.
      inputText = inputText.replace(/\b[a-z]/i, $1 => $1.toUpperCase())
    }

    return inputText
  }
```

I haven't made this into a commit yet. Additionally, I was not sure where to place it. I initially placed it in `discard-changes-dialog.tsx` near where it's needed for testing purposes. However, if we want this as a global helper function, I was not exactly confident where it should go.

A somewhat similar bit of code I have bumped into before was the [`src/lib/sanitize-branch.ts`](https://github.com/desktop/desktop/blob/master/app/src/lib/sanitize-branch.ts) which gets exported out. If we were to place it in `/src/lib/` I could additionally add a test for this function.

But that's definitely jumping ahead a bit as there may be changes needed, and decide if we'll continue with this function.

Reply:
this solution looks great to me! and i agree /src/lib/ is a good place for it, probably in its own file for now. and a test would be amazing! full steam ahead :)

Reply:
Thanks for these excellent notes!

> A test for this comes with some additional difficulties

I forgot about that. :) What i've done in the past for is something like this is:

```typescript
    it('does something good', () => {
      if (__WIN32__) {
        // Sentence case expect
      }
      if (__DARWIN__ || __LINUX__) {
        // Title case expect
      }
    })
```

Not the prettiest solution, so if you have another approach I'm all ears!

> It appears that @j-f1 has worked on a much better solution for this problem

Thanks for finding this! I didn't know it existed. There's a lot of good points about limitations and while I think that the solution there is an excellent comprehensive solution, I would say this is still a good incremental step to cover the majority of cases. I'd rather get this in in the meantime.

Thank you again for all your work on this @Daniel-McCarthy! I just wrote a component today that I'm planning to use this new helper in (twice!).

Reply:
Excellent, I have implemented the test as suggested by combining the platform test into one. 

>I'd rather get this in in the meantime.

Sounds great, I will have this ready very soon and update it if there any problems with building.

>Thank you again for all your work on this

No problem, I'm happy to help! I'm glad it will be useful.